boards/arduino-mega2560/board.c

```diff
@@ -56,7 +56,7 @@ void board_init(void)
 void SystemInit(void)
 {
     /* initialize UART_0 for use as stdout */
```

Comment:
It looks like not all the arduino platforms are actually dependent on the arduino-common board right now.  

| **Board** |  Includes arduino-common | Has STDIO implementation|
| ---- | ---- | ----- | 
| arduino-due | NO | NO |
| arduino-duemilanove | YES | NO |
| arduino-mega2560 | NO | YES |
| arduino-uno | YES | NO |
| arduino-zero | NO | NO |

Based on the comment from arduino-common/Makefile.include below, I think that framework was only inteneded for uno and duemilanove boards

```
# define the cpu used by the arduino uno and duemilanove boards
export CPU = atmega328p
```

Let me know what you want me to do here.